Sentry.php

```diff
@@ -0,0 +1,134 @@
+ * Copyright (C) 2011 - 2019 SalesAgility Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License version 3 as published by the
+ * Free Software Foundation with the addition of the following permission added
+ * to Section 15 as permitted in Section 7(a): FOR ANY PART OF THE COVERED WORK
+ * IN WHICH THE COPYRIGHT IS OWNED BY SUGARCRM, SUGARCRM DISCLAIMS THE WARRANTY
+ * OF NON INFRINGEMENT OF THIRD PARTY RIGHTS.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program; if not, see http://www.gnu.org/licenses or write to the Free
+ * Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+class Sentry
+{
+    protected static $enabled;
+    protected static $dsn;
+    protected static $instance;
+    protected static $client;
+
+    public static function getInstance()
+    {
+        if (self::$instance !== null) {
+            return self::$instance;
+        }
+
+        return self::$instance = new self();
+    }
+
+    public function __construct()
+    {
+        global $sugar_config;
+
+        try {
+            if (!isset($sugar_config['sentry']['dsn'])) {
+                throw new \SuiteCRM\Exception\Exception('You must configure Sentry DSN.');
+            }
+        } catch (\SuiteCRM\Exception\Exception$exception) {
+            LoggerManager::getLogger()->fatal($exception->getMessage() . "\nTrace:\n" . $exception->getTraceAsString());
+        }
+
+        self::$enabled = $sugar_config['sentry']['enabled'];
+        self::$dsn = $sugar_config['sentry']['dsn'];
+        self::$client = $this->buildClient();
+        $this->init();
+    }
+
+    protected function buildClient()
+    {
+        return new Raven_Client(self::$dsn);
```

Comment:
As an escape hatch in case the user wants to customize something without waiting for a SuiteCRM update to add it, we should probably have a `$sugar_config['sentry']['extra_config']` array that'd be passed as the second argument to Raven_Client.

https://docs.sentry.io/clients/php/config/